script.rb

```diff
@@ -0,0 +1,140 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+# rubocop: disable Lint/ToEnumArguments
+
+module Enumerable
+  # my_each
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i]
+      i += 1
+    end
+    self
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    x = to_a if to_a.is_a? Range
+    x = self if to_a.is_a? Array
+    i = 0
+    while i < x.length
+      yield x[i], i
+      i += 1
+    end
+    self
+  end
+
+  # my_select
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    my_arr = []
+    to_a.my_each do |num|
+      my_arr.push(num) if yield num
+    end
+    my_arr
+  end
+
+  # my_all
+  def my_all?(param = nil)
+    if !block_given? && param.nil?
+      to_a.my_each { |num| return false if num == false || num.nil? }
+    elsif !block_given? && (param.is_a? Class)
+      to_a.my_each { |item| return false if item.class != param }
+    elsif !block_given? && (param.is_a? Regexp)
+      to_a.my_each { |items| return false unless items.match(param) }
+    elsif block_given?
+      to_a.my_each { |num| return false unless yield num }
+    end
+    true
+  end
```

Comment:
- [ ] returns true if all of the collection is a member of such class::Numeric
```
  p [1,2,3,4].all?(Numeric)
  p [1,2,3,4].my_all?(Numeric) 

# This should return true
```

- [ ]  when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern

```
  p [3,3,3,3, 2].all?(3) // false
  p [3,3,3,3,2].my_all?(3) // true

# This should be false
```